tests/test_base_plugin.py

```diff
@@ -0,0 +1,282 @@
+        panoptes_plugin_info = PanoptesPluginInfo("plugin_name", "path/to/plugin")
+        panoptes_plugin_info.panoptes_context = self.__panoptes_context
+        panoptes_plugin_info.kv_store_class = PanoptesTestKeyValueStore
+        panoptes_plugin_info.config_filename = "tests/plugins/polling/test/plugin_polling_test.panoptes-plugin.example"
+        panoptes_plugin_info.details.read(panoptes_plugin_info.config_filename)
+
+        mock_moduleMtime = _TIMESTAMP - 1
+        mock_configMtime = _TIMESTAMP - 2
+        with patch('yahoo_panoptes.framework.plugins.panoptes_base_plugin.PanoptesPluginInfo.configMtime',
+                   mock_configMtime):
+            with patch('yahoo_panoptes.framework.plugins.panoptes_base_plugin.PanoptesPluginInfo.moduleMtime',
+                       mock_moduleMtime):
+                # Ensure first if-block in execute_now returns False
+                panoptes_plugin_info.last_results = int(_TIMESTAMP)
+                panoptes_plugin_info.last_executed = int(_TIMESTAMP)
+                self.assertFalse(panoptes_plugin_info.execute_now)
+
+    def test_plugin_info_execute_now_2(self):
+        panoptes_plugin_info = PanoptesPluginInfo("plugin_name", "path/to/plugin")
+        panoptes_plugin_info.panoptes_context = self.__panoptes_context
+        panoptes_plugin_info.kv_store_class = PanoptesTestKeyValueStore
+        panoptes_plugin_info.config_filename = "tests/plugins/polling/test/plugin_polling_test.panoptes-plugin.example"
+        panoptes_plugin_info.details.read(panoptes_plugin_info.config_filename)
+
+        mock_moduleMtime = _TIMESTAMP - 1
+        mock_configMtime = _TIMESTAMP - 2
+        with patch('yahoo_panoptes.framework.plugins.panoptes_base_plugin.PanoptesPluginInfo.configMtime',
+                   mock_configMtime):
+            with patch('yahoo_panoptes.framework.plugins.panoptes_base_plugin.PanoptesPluginInfo.moduleMtime',
+                       mock_moduleMtime):
+                # Ensure second if-block in execute_now returns False
+                panoptes_plugin_info.last_results = int(_TIMESTAMP)
+                panoptes_plugin_info.last_executed = (int(_TIMESTAMP) - panoptes_plugin_info.execute_frequency)
+                self.assertFalse(panoptes_plugin_info.execute_now)
+
+    def test_plugin_info_execute_now_3(self):
+        panoptes_plugin_info = PanoptesPluginInfo("plugin_name", "path/to/plugin")
+        panoptes_plugin_info.panoptes_context = self.__panoptes_context
+        panoptes_plugin_info.kv_store_class = PanoptesTestKeyValueStore
+        panoptes_plugin_info.config_filename = "tests/plugins/polling/test/plugin_polling_test.panoptes-plugin.example"
+        panoptes_plugin_info.details.read(panoptes_plugin_info.config_filename)
+
+        mock_moduleMtime = _TIMESTAMP - 1
+        mock_configMtime = _TIMESTAMP - 2
+        with patch('yahoo_panoptes.framework.plugins.panoptes_base_plugin.PanoptesPluginInfo.configMtime',
```

Comment:
How about we merge this with  test_plugin_info_execute_now since we can avoid the duplicate setup code